applications/StructuralMechanicsApplication/custom_advanced_constitutive/wrinkling_linear_2d_law.h

```diff
@@ -0,0 +1,393 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Klaus B. Sautter
+//
+
+#if !defined (KRATOS_WRINKLING_LINEAR_2D_LAW_H_INCLUDED)
+#define  KRATOS_WRINKLING_LINEAR_2D_LAW_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/constitutive_law.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+/**
+ * @class WrinklingLinear2DLaw
+ * @ingroup StructuralMechanicsApplication
+ * @brief This law defines a wrinkling modification for any 2D claw
+ * @author Klaus B. Sautter
+ */
+class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) WrinklingLinear2DLaw
+    : public ConstitutiveLaw
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// The definition of the process info
+    typedef ProcessInfo ProcessInfoType;
+
+    /// The definition of the CL base  class
+    typedef ConstitutiveLaw    BaseType;
+
+    /// The definition of the size type
+    typedef std::size_t        SizeType;
+
+    /// Pointer definition of WrinklingLinear2DLaw
+    KRATOS_CLASS_POINTER_DEFINITION( WrinklingLinear2DLaw );
+
+    enum class WrinklingType {
+      Taut,
+      Slack,
+      Wrinkle
+    };
+
+    ///@name Lyfe Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor.
+     */
+    WrinklingLinear2DLaw();
+
+    /**
+     * @brief Copy constructor.
+     */
+    WrinklingLinear2DLaw (const WrinklingLinear2DLaw& rOther);
+
+    /**
+     * @brief Destructor.
+     */
+    ~WrinklingLinear2DLaw() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Clone operation
+     */
+    ConstitutiveLaw::Pointer Clone() const override;
+
+    /**
+     * @brief Creates a new constitutive law pointer
+     * @param NewParameters The configuration parameters of the new constitutive law
+     * @return a Pointer to the new constitutive law
+     */
+    ConstitutiveLaw::Pointer Create(Kratos::Parameters NewParameters) const override;
+
+    /**
+     * @brief Dimension of the law
+     * @details This is not used, so 0 is returned
+     */
+    SizeType WorkingSpaceDimension() override;
+
+    /**
+     * @brief Voigt tensor size
+     * @details This is not used, so 0 is returned
+     */
+    SizeType GetStrainSize() override;
+
+
```

Comment:
Docimentation is missing

Reply:
documentation for virtual functions?

Reply:
IMO

Reply:
alright, so just copy the documentation of the virtual function?

Reply:
yes